Use ringbuffer and sched_process_exec tracepoint

allir opened the pull request:
* The `sys_enter_exeve` tracepoint has some issues on arm64 with finding the filename of the process so switching to `sched_process_exec`
* Use ringbuffer instead of perf array buffer for the eBPF Map.
* Fix minor issues in Makefile.

arnarpall commented on main.go:
```diff
@@ -21,12 +22,12 @@ func main() {
 
-	execve := &ebpf.Execve{}
-	must(execve.Start(), "execve start")
+	processExec := &ebpf.ProcessExecTracePoint{}
```
let`s use empty semantics here.
processExec := new(ebpf.ProcessExecTracePoint)

emanela commented on bpf/program.bpf.c:
```diff
@@ -1,62 +1,86 @@
 
-	bpf_perf_event_output(args, &events,
-		BPF_F_CURRENT_CPU, &exec_data, sizeof(exec_data));
+	bpf_ringbuf_output(&events, e, sizeof(*e), 0);
```
If you are using the fixed size of the struct you can utilize the reserve ringbuff and avoid the heap usage.

allir replied:
Ah, nice, I'll update to that.

allir committed:
Use ringbuffer and sched_process_exec tracepoint

The `sys_enter_exeve` tracepoint has some issues on arm64 with finding the
filename of the process so switching to `sched_process_exec`

Use ringbuffer instead of perf array buffer for the eBPF Map.
